extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/SyntheticBeanBuildItem.java

```diff
@@ -181,6 +192,19 @@ public ExtendedBeanConfigurator setRuntimeInit() {
             return this;
         }
 
+        /**
+         * The {@link #checkActive(Consumer)} procedure is a {@link Runnable} proxy returned from a recorder method.
+         *
+         * @param checkActive a {@code Runnable} returned from a recorder method
+         * @return self
+         * @throws IllegalArgumentException If the {@code checkActive} argument is not a proxy returned from a recorder method
+         */
+        public ExtendedBeanConfigurator checkActive(Runnable checkActive) {
```

Comment:
It might make sense to add a variant that would accept a bytecode generator, similarly to `io.quarkus.arc.processor.BeanConfiguratorBase.creator(Consumer<MethodCreator>)`.

Reply:
There's `BeanConfiguratorBase.checkActive(Consumer<MethodCreator>)`, so I think that's already present.